core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1486,13 +1438,15 @@ class KafkaApis(val requestChannel: RequestChannel,
 
     val shareFetchData = shareFetchRequest.shareFetchData(topicNames)
     val forgottenTopics = shareFetchRequest.forgottenTopics(topicNames)
+    var cachedTopicPartitions : util.List[TopicIdPartition] = null
 
-    var cachedTopicPartitions : util.List[TopicIdPartition] = new util.ArrayList[TopicIdPartition]
-    try {
-      cachedTopicPartitions = sharePartitionManager.cachedTopicIdPartitionsInShareSession(groupId, Uuid.fromString(memberId))
-    } catch {
+    if (shareSessionEpoch == ShareFetchMetadata.FINAL_EPOCH) {
```

Comment:
I don't quite see what's special about FINAL_EPOCH here.

Reply:
Thanks for reviewing the PR. We actually don't need the variable `cachedTopicPartitions` at all until it's a FINAL request. That's why the check was added here. When it's the final epoch and the consumer does not acknowledge all the acquired records, then the unacknowledged but acquired records need to be released. cachedTopicPartitions gives a list of all the topic partitions from which the consumer consumed. Later in the end of the function, this list is used to release all acquired records from the same memberId in all these topic partitions.

Reply:
I think this is worth a comment. You can do that on another commit in this PR, or a future PR.